tardis/rest/app/routers/user.py

```diff
@@ -0,0 +1,92 @@
+from ..scopes import User
+from .. import security
+from fastapi import APIRouter, Depends, Security
+from fastapi_jwt_auth import AuthJWT
+
+router = APIRouter(prefix="/user", tags=["user"])
+
+
+@router.post(
+    "/login",
+    description="Sets httponly access token in session cookie. The scopes are optional.",  # noqa B950
+)
+async def login(
+    login_user: security.LoginUser,
+    Authorize: AuthJWT = Depends(),
+):
+    user = security.check_authentication(login_user.user_name, login_user.password)
+
+    if login_user.scopes is None:
+        scopes = {"scopes": user.scopes}
+    else:
+        # The next two lines are very critical as if wrongly implemented a user can give his token unlimited scopes. # noqa B950
+        # This functionality has to be tested thoroughly
+        security.check_scope_permissions(login_user.scopes, user.scopes)
+        scopes = {"scopes": login_user.scopes}
+
+    access_token = Authorize.create_access_token(
+        subject=user.user_name, user_claims=scopes
+    )
+    refresh_token = Authorize.create_refresh_token(
+        subject=user.user_name, user_claims=scopes
+    )
+
+    Authorize.set_access_cookies(access_token)
+    Authorize.set_refresh_cookies(refresh_token)
+
+    # It is ok to return the user here because every user should have the user:get scope. # noqa B950
+    # Tokens don't have permissions to this function
```

Comment:
How is this secured?

Reply:
This function isn't accessible to tokens. Thus only users with database username and password can login. It made sense to me that a real user should be able to request information about himself without the user:get scope. In hindsight this might not have been the best decision. I will put a guard here I you agree.

Reply:
I added a guard